File: gaia-sdk-python/gaia_sdk/graphql/request/type/RoleKeyOne.py
from gaia_sdk.api.VariableRegistry import VariableRegistry


class RoleKeyOne(list):

    def tenant_id(self):
        self.append(lambda x: "tenantId")

    def role_id(self):
        self.append(lambda x: "roleId")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/RoleKeyOne.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class RoleKeyOne extends Array<(_:VariableRegistry) => string> {
public _typeName = "RoleKeyOne";
    public tenantId = () => { 
        this.push(_ => "tenantId")
    };

    public roleId = () => { 
        this.push(_ => "roleId")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

